content/docs/1.2.4/volumes-and-nodes/volume-size.md

```diff
@@ -19,29 +19,69 @@ In this section, you'll have a better understanding of concepts related to volum
-    - There is some meta info in the volume hence the `actual size` is not exactly 0.
 2. Write 2Gi data (data#1) in the volume mount point and take a snapshot (snapshot#1). See Figure 2 of the illustration.
     - Now data#1 is stored in snapshot#1 and the `actual size` is 2Gi.
 {{< figure src="/img/screenshots/volumes-and-nodes/volume-size-example-1.png" >}}
 3. Delete data#1 from the mount point.
-    - The truth of the data#1 deletion is that the data#1 is marked as removed in **the filesystem level** (For example, inode deletion in ext4). Since **Longhorn operates on the block level** and does not understand the filesystem, as a result, the disk blocks/space storing data#1 won't be released after the deletion.
     - The data#1 filesystem level removal info is stored in current volume head file. For snapshot#1, data#1 is still retained as the historical data.
     - The `actual size` is still 2Gi.
 {{< figure src="/img/screenshots/volumes-and-nodes/volume-size-example-2.png" >}}
+
 4. Write 4Gi data (data#2) in the volume mount, then take one more snapshot (snapshot#2). See Figure 3 of the illustration.
     - Now the `actual size` is 6Gi, which is greater than the nominal `size`.
     - There is overlapping between 2 snapshots in block level (See the 2 snapshots in Figure 3), since the data#1 is marked as removed in snapshot#2 then the filesystem reuses the space.
+
 {{< figure src="/img/screenshots/volumes-and-nodes/volume-size-example-3.png" >}}
 5. Delete snapshot#1 and wait for snapshot purge complete. See Figure 4 of the illustration.
     - Here Longhorn actually coalesces the snapshot#1 with the snapshot#2.
     - For the overlapping part during the coalesce, the newer data (data#2) will be retained in the blocks. Then some historical data is removed and the volume gets shrunk (from 6.1Gi to 4.65Gi in the example).
 {{< figure src="/img/screenshots/volumes-and-nodes/volume-size-example-4.png" >}}
+
+
+
+## Space Configuration Suggestions for Volumes
+
+1. Expand disks in the nodes so that there is an enough buffer for Longhorn volumes.
+    - Each snapshot or volume head contains at most `size` space. A quick esitimation for maximum space consumption estimation of a volume is
+  
+        ```
+        maximum space consumption = (N + 1) x volume nominal size
+        ```
+
+        where N is the number of snapshots retained, and 1 is for the system snapshot.
+
+2. Do not retain too many snapshots for the volumes.
+
+3. Clean up snapshots will help reclaim disk space. There are two ways to clean up snapshots:
+    - Delete the snapshots manually via Longhorn UI.
+    - Set a snapshot recurring job with retention 1, then the snapshots will be cleaned up automatically.
+ 
+    Also, notice that the extra space, up to volume nominal `size`, is required during snapshot cleanup and merge.
+
+4. Set `Storage Over Provisioning Percentage` to be smaller than 100% if there are write-heavy volumes.
```

Comment:
Maybe you can combine 1st item with 4th item.
- Reserve enough free space in disks as buffers in case of the actual size of existing volumes keep growing up. 
  - A quick estimation for maximum space consumption estimation of a volume is: 
    ```
    (N + 1 + 1) x head/snapshot average actual size
    ```
    where N is the number of snapshots retained, and the 1st 1 is for the system snapshot, and 2nd 1 is for the tmp space may required by snapshot deletion. 
  - If there are heavy writing tasks for volumes, `head/snapshot average actual size` would be volume the nominal size. And in this case, it's better to set `Storage Over Provisioning Percentage` to be smaller than 100%
to avoid disk space exhaustion.